src/engine/SpatialJoin.cpp

```diff
@@ -152,6 +113,27 @@ string SpatialJoin::getCacheKeyImpl() const {
     if (maxResults.has_value()) {
       os << "maxResults: " << maxResults.value() << "\n";
     }
+
+    // Uses distance variable?
+    if (config_->distanceVariable_.has_value()) {
+      os << "withDistanceVariable\n";
+    }
+
+    // Selected payload variables
+    os << "payload:";
+    if (std::holds_alternative<PayloadAllVariables>(
+            config_->payloadVariables_)) {
+      os << " AllVariables";
+    } else {
+      auto vec = std::get<std::vector<Variable>>(config_->payloadVariables_);
+      for (auto v : vec) {
+        os << " " << v.name();
+      }
+    }
```

Comment:
You are right, this was also a bug in the previous implementation: if a child contains more than one col with point geometries and we change only the col used for the join between two queries, the old (=incorrect) result from the cache will be fetched instead of the correct one.